dvc/tree/dvc.py

```diff
@@ -224,6 +238,19 @@ def isdvc(self, path, recursive=False, strict=True):
     def isexec(self, path):  # pylint: disable=unused-argument
         return False
 
+    def get_dir_hash(self, path_info, **kwargs):
+        try:
+            outs = self._find_outs(path_info, strict=True)
+            if len(outs) == 1 and outs[0].is_dir_checksum:
+                out = outs[0]
+                # other code expects us to fetch the dir at this point
```

Comment:
We could do the same thing for a subdir in a cached dir, but instead we just fallback to `super().get_dir_hash` since it is simpler. This will be cleaned up a bit more after https://github.com/iterative/dvc/issues/4144